drivers/w1/w1_net_intern.c

```diff
@@ -0,0 +1,203 @@
+ * the discrepancy is set to W1_SEARCH_NO_DEVICE.
+ *
+ * The Implementation is similar as suggested in the maxim application note 187.
+ * The controller reads the first rom bit and its complementary value of all
+ * devices.
+ * Due to physical characteristics, the value received is a
+ * logical AND of all devices 1st bit. Devices only continue to participate
+ * in the search procedure if the next bit the controller sends matches their
+ * own addresses bit. This allows the controller to branch through 64-bit
+ * addresses in order to detect all devices.
+
+ * The 1st bit received is stored in bit 1 of rom_inv_64, the 2nd in bit 2 and so
+ * on until bit 64.
+ * As a result each byte of the rom has the correct bit order, but the received
+ * bytes (big-endian) stored in rom_inv_64 are in inverse byte order.
+ *
+ * Note: Filtering on families is not supported.
+ *
+ * @param dev              Pointer to the device structure for the w1 instance.
+ * @param command          Command to chose between normal and alarm search.
+ * @param family           The family is ignored in search.
+ * @param last_discrepancy This must be set to W1_SEARCH_DISCREPANCY_INIT before
+ *                         the first call, it carries the search progress for
+ *                         further calls.
+ * @param rom_inv_64       The found rom: It must be set to zero before first
+ *                         call and carries the last found rom for furter calls.
+ *                         The rom is stored in inverse byte order.
+ *
+ * @retval 0 If successful.
+ * @retval -EIO in case of 1-wire read/write error
+ */
+static int search_device(const struct device *dev, uint8_t command,
+			 uint8_t family, size_t *last_discrepancy,
+			 uint64_t *rom_inv_64)
+{
+	int ret;
+	size_t next_discrepancy;
+
+	ARG_UNUSED(family);
+	__ASSERT_NO_MSG(command == W1_CMD_SEARCH_ROM ||
+			command == W1_CMD_SEARCH_ALARM);
+
+	ret = w1_reset_bus(dev);
+	if (ret < 0) {
+		return -EIO;
+	}
+	if (ret == 0) {
+		*last_discrepancy = W1_SEARCH_NO_DEVICE;
+		return 0;
+	}
+
+	ret = w1_write_byte(dev, command);
+	if (ret < 0) {
+		return -EIO;
+	}
+	next_discrepancy = W1_SEARCH_LAST_DEVICE;
+
+	for (size_t id_bit_nr = 1; id_bit_nr < 65; id_bit_nr++) {
+		bool last_id_bit;
+		bool last_complement_id_bit;
+
+		ret = w1_read_bit(dev);
+		if (ret < 0) {
+			return -EIO;
+		}
+		last_id_bit = (bool)ret;
+		ret = w1_read_bit(dev);
+		if (ret < 0) {
+			return -EIO;
+		}
+		last_complement_id_bit = (bool)ret;
+
+		if (last_id_bit && last_complement_id_bit) {
+			/*
+			 * No device participating:
+			 * We can stop following the branch.
+			 */
+			LOG_DBG("No device paricipating");
+			*last_discrepancy = W1_SEARCH_NO_DEVICE;
+			return 0;
+		} else if (last_id_bit != last_complement_id_bit) {
+			/*
+			 * All devices connected have same ROM bit value:
+			 * We can directly follow last_id_bit branch.
+			 */
+		} else {
+			/*
+			 * (!last_id_bit && !last_complement_id_bit)
```

Comment:
?

Reply:
This comments only the the condition for taking this branch, so it is really not needed. I will remove it.